tools/setup.sh

```diff
@@ -1,32 +1,58 @@
 #!/usr/bin/env bash
 
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+UNDERLINE='\033[4m'
+BOLD='\033[1m'
+NC='\033[0m'
+
 set -e
 
-if [ ! -f launch_openpilot.sh ]; then
-  if [ ! -d openpilot ]; then
-    git clone --single-branch --recurse-submodules https://github.com/commaai/openpilot.git
-  fi
-  cd openpilot
+if [ -z "$OPENPILOT_ROOT" ]; then
+  # default to current directory for installation
+  OPENPILOT_ROOT="$(pwd)/openpilot"
 fi
 
-if [[ "$OSTYPE" == "darwin"* ]]; then
-  tools/mac_setup.sh
-else
-  tools/ubuntu_setup.sh
-fi
+function check_dir() {
+  echo "Checking for installation directory..."
+  if [ -d "$OPENPILOT_ROOT" ]; then
+    echo -e " ↳ [${RED}✗${NC}] can't install openpilot in $OPENPILOT_ROOT !"
+    return 1
+  fi
 
-git lfs pull
+  echo -e " ↳ [${GREEN}✔${NC}] Successfully chosen $OPENPILOT_ROOT as installation directory\n"
+}
+
+function check_git() {
+  echo "Checking for git..."
+  if ! command -v "git" > /dev/null 2>&1; then
+    echo -e " ↳ [${RED}✗${NC}] git not found on your system, can't continue!"
+    return 1
+  else
+    echo -e " ↳ [${GREEN}✔${NC}] git found.\n"
+  fi
+}
+
+function git_clone() {
+  echo "Cloning openpilot..."
+  if $(git clone --depth=1 https://github.com/commaai/openpilot.git "$OPENPILOT_ROOT"); then
```

Comment:
since the cloning is in `--depth 1` (shallow) it should be trivial to:
- add your own remote (fork) and unshallow - might work by explaining something like [this](https://stackoverflow.com/a/17714718/639708) and `git fetch --unshallow`
- if I say "N" to "macOS support" it should continue to git submodules and lfs (works fine when saying "Y")

Reply:
depth 1 was mostly for testing speed. If you think that those commands should be added to op, feel free to open a pr or discuss it on the discord, but general I don't think we should add git command to op unless it is very related to an openpilot task. The fork one might be good, but I don't think we should include the unshallow one, that is way to git specific